Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -501,80 +491,76 @@ private void HandleTwoHandManipulationUpdated()
+                targetRotation = Quaternion.LookRotation(-directionToTarget);
+            }
+            else if (rotateInOneHandType == RotateInOneHandType.FaceAwayFromUser)
+            {
+                Vector3 directionToTarget = hostTransform.position - CameraCache.Main.transform.position;
+                targetRotation = Quaternion.LookRotation(directionToTarget);
+            }
+            else
+            {
+                targetRotation = pointer.Rotation * objectToHandRotation;
+                switch (constraintOnRotation)
                 {
-                    targetRotation = interactionMapping.PoseData.Rotation * objectToHandRotation;
-                    switch (constraintOnRotation)
-                    {
-                        case RotationConstraintType.XAxisOnly:
-                            targetRotation.eulerAngles = Vector3.Scale(targetRotation.eulerAngles, Vector3.right);
-                            break;
-                        case RotationConstraintType.YAxisOnly:
-                            targetRotation.eulerAngles = Vector3.Scale(targetRotation.eulerAngles, Vector3.up);
-                            break;
-                        case RotationConstraintType.ZAxisOnly:
-                            targetRotation.eulerAngles = Vector3.Scale(targetRotation.eulerAngles, Vector3.forward);
-                            break;
-                    }
+                    case RotationConstraintType.XAxisOnly:
+                        targetRotation.eulerAngles = Vector3.Scale(targetRotation.eulerAngles, Vector3.right);
+                        break;
+                    case RotationConstraintType.YAxisOnly:
+                        targetRotation.eulerAngles = Vector3.Scale(targetRotation.eulerAngles, Vector3.up);
+                        break;
+                    case RotationConstraintType.ZAxisOnly:
+                        targetRotation.eulerAngles = Vector3.Scale(targetRotation.eulerAngles, Vector3.forward);
+                        break;
                 }
+            }
 
-                Vector3 targetPosition;
-                if (IsNearManipulation())
+            Vector3 targetPosition;
+            if (IsNearManipulation())
+            {
+                if (oneHandRotationModeNear == RotateInOneHandType.RotateAboutGrabPoint)
                 {
-                    if (oneHandRotationModeNear == RotateInOneHandType.RotateAboutGrabPoint)
-                    {
-                        targetPosition = (interactionMapping.PoseData.Rotation * objectToHandTranslation) + interactionMapping.PoseData.Position;
-                    }
-                    else // RotateAboutCenter or DoNotRotateInOneHand
-                    {
-                        targetPosition = objectToHandTranslation + interactionMapping.PoseData.Position;
-                    }
+                    targetPosition = (pointer.Rotation * objectToHandTranslation) + pointer.Position;
```

Comment:
Will this correctly use the grip pose rotation instead of the pointer pose rotation?

Reply:
I believe it shouldn't matter because we're only interested in how the rotation changes over time, not the absolute value. In practice, it seems to work fine for 6 DOF controllers and near interaction but hand rays don't produce any rotation.